synchronization/src/main/java/de/cyface/synchronization/HttpConnection.java

```diff
@@ -185,37 +192,41 @@ public HttpResponse post(@NonNull final HttpURLConnection connection, @NonNull f
             throws SynchronisationException, BadRequestException, UnauthorizedException, InternalServerErrorException,
             ForbiddenException, EntityNotParsableException, ConflictException, NetworkUnavailableException,
             SynchronizationInterruptedException, TooManyRequestsException {
 
-        // Generate header
+        // Generate MetaData Multipart header
         // Attention: Parts of the header (Content-Type, boundary, request method, user agent) are already set
-        final String remainingHeader = generateHeader(metaData, fileName);
+        final String remainingHeader = generateHeader(metaData);
         final byte[] remainingHeaderBytes = remainingHeader.getBytes();
 
-        // Set the fixed number of bytes which will be written to the OutputStream
-        final long binarySize = transferTempFile.length();
-        Validate.isTrue(binarySize != 0L); // We don't post empty files, ensure files still exists
-        final long fixedStreamLength = calculateBytesWrittenToOutputStream(remainingHeaderBytes, binarySize);
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
-            connection.setFixedLengthStreamingMode(fixedStreamLength);
-        } else {
-            connection.setFixedLengthStreamingMode((int)fixedStreamLength);
-        }
-        // connection.setRequestProperty("Content-length" should be obsolete with setFixedLengthStreamingMode
+        // Generate File Multipart headers
+        final String measurementFileHeaderPart = generateFileHeaderPart("fileToUpload", fileName);
+        final String eventsFileHeaderPart = generateFileHeaderPart("eventsFile", eventsFileName);
+
+        // The streaming length needs to be set up before the connection is connected.
+        final long fixedStreamLength = setupFixedLengthStreamingMode(connection, remainingHeaderBytes.length,
+                measurementTransferTempFile.length(), eventsTransferTempFile.length(),
+                measurementFileHeaderPart.getBytes().length,
+                eventsFileHeaderPart.getBytes().length);
 
         // Use a buffered stream to upload the transfer file to avoid OOM and for performance
         final FileInputStream fileInputStream;
+        final FileInputStream eventFileInputStream;
         try {
-            fileInputStream = new FileInputStream(transferTempFile);
+            fileInputStream = new FileInputStream(measurementTransferTempFile);
+            eventFileInputStream = new FileInputStream(eventsTransferTempFile);
         } catch (final FileNotFoundException e) {
             throw new IllegalStateException(e);
         }
-        final BufferedInputStream bufferedFileInputStream = new BufferedInputStream(fileInputStream);
+        final BufferedInputStream bufferedMeasurementFileInputStream = new BufferedInputStream(fileInputStream);
+        final BufferedInputStream bufferedEventsFileInputStream = new BufferedInputStream(eventFileInputStream);
```

Comment:
Are these InputStreams closed somewhere (ideally in a finally block)? Otherwise this is a memory leak.